### PR TITLE
[presentation-api] add a step to terminate presentations when the test stops

### LINKED ISSUE #4258: Add a cleanup step to terminate presentations
Given that presentations may persist beyond the controlling browsing context, tests should always terminate presentations that they create (even when the test fails), typically using the [`add_cleanup` method](https://github.com/w3c/testharness.js/blob/master/docs/api.md#cleanup).

Examples of Presentation API tests that need this:
https://github.com/w3c/web-platform-tests/blob/master/presentation-api/controlling-ua/startNewPresentation_success-manual.html
https://github.com/w3c/web-platform-tests/blob/master/presentation-api/controlling-ua/reconnectToPresentation_success-manual.html

### PR DESCRIPTION
**Pull request by tomoyukilabs**
This PR partially fixes #4258.

* `startNewPresentation_unsettledpromise-manual.html` now terminates a presentation in `request.start().then()`, since the display selection dialog could appear after the test result would be shown.
* The other tests now terminate a presentation in `t.add_cleanup(function)`.

### CODE REVIEW

File: presentation-api/controlling-ua/startNewPresentation_displaynotallowed-manual.html [hunk 1]
```diff
@@ -21,11 +24,19 @@
...
         promise_test(function (t) {
             var request = new PresentationRequest(presentationUrls);
-            return promise_rejects(t, 'NotAllowedError', request.start());
+
+            // terminate the presentation connection when the presentation is started by accident
+            var connection;
+            t.add_cleanup(function() {
+                if(connection)
+                    connection.terminate();
+            })
```
Comment (tidoust) [inline presentation-api/controlling-ua/startNewPresentation_displaynotallowed-manual.html]: Just for the sake of code consistency, do you mean adding back the missing ";" here?

File: presentation-api/controlling-ua/startNewPresentation_displaynotfound-manual.html [hunk 2]
```diff
@@ -21,11 +24,19 @@
...
         promise_test(function (t) {
             var request = new PresentationRequest(presentationUrls);
-            return promise_rejects(t, 'NotFoundError', request.start());
+
+            // terminate the presentation connection when the presentation is started by accident
+            var connection;
+            t.add_cleanup(function() {
+                if(connection)
+                    connection.terminate();
+            })
```
Comment (tidoust) [inline presentation-api/controlling-ua/startNewPresentation_displaynotfound-manual.html]: Same here, do you mean adding back the missing ";"?

**Review by tidoust — CHANGES_REQUESTED** [review]
That looks great. See trivial comments inline about missing ";" for code consistency.

**Comment by tomoyukilabs** [comment]
Oops. I've added missing semicolon.

**Pull request merged**